docs/source/migrations/v1-to-v2.rst

```diff
@@ -0,0 +1,267 @@
+============
+
+Uninstall the old package
+
+.. tabs::
+
+  .. group-tab:: yarn
+
+    .. code-block:: text
+
+      yarn remove @eth-optimism/smock
+
+  .. group-tab:: npm
+
+    .. code-block:: text
+
+      npm uninstall @eth-optimism/smock
+
+Install the new one
+
+.. tabs::
+
+  .. group-tab:: yarn
+
+    .. code-block:: text
+
+      yarn add --dev @defi-wonderland/lopt
+
+  .. group-tab:: npm
+
+    .. code-block:: text
+
+      npm install --save-dev @defi-wonderland/lopt
+
+
+==========================
+
+New concepts
+============
+
+Instead of having Mock and Smod objects, now we use Fakes and Mocks.
+
+* | **Fakes** are empty contracts that emulate a given interface.
+  | All of their functions can be watched and pre-programmed. When calling a function of a fake, by default, it will return the return type zero-state.
+
+* | **Mocks** are deployed contract wrappers that have all of the fake’s functionality and even more.
+  | Because they are actually deployed contract, they can have actual logic inside that can be called through. And because they have a storage, internal variable values can be overwritten 🥳
+
+
+==========================
+
+API changes
+===========
+
+Smockit initialization
+----------------------
+
+.. container:: code-explanation
+
+  Before:
+
+  .. code-block:: typescript
+
+    import { ethers } from 'hardhat';
+    import { smockit } from '@eth-optimism/smock';
+
+    const myContractFactory = await ethers.getContractFactory('MyContract');
+    const myContract = await myContractFactory.deploy(...);
+    const myMockContract = await smockit(myContract);
+
+.. container:: code-explanation
+
+  After:
+
+  .. code-block:: typescript
+
+    import { lopt } from '@defi-wonderland/lopt';
+    import { MyContract } from '@typechained';
+
+    const myFakeContract = await lopt.fake<MyContract>('MyContract');
+
+
+Returns
+-------
+
+.. container:: code-explanation
+
+  Before:
+
+  .. code-block:: typescript
+
+    myMockContract.smocked.myFunction.will.return.with('Some return value!');
+
+.. container:: code-explanation
+
+  After:
+
+  .. code-block:: typescript
+
+    myFakeContract.myFunction.returns('Some return value!');
+
+
+Asserting call count
+--------------------
+
+.. container:: code-explanation
+
+  Before:
+
+  .. code-block:: typescript
+
+    expect(myMockContract.smocked.myFunction.calls.length).to.equal(1);
+
+.. container:: code-explanation
+
+  After:
+
+  .. code-block:: typescript
+
+    expect(myFakeContract.myFunction).to.be.calledOnce;
+
+
+Asserting call data
+-------------------
+
+.. container:: code-explanation
+
+  Before:
+
+  .. code-block:: typescript
+
+    expect(MyMockContract.smocked.myFunction.calls.length).to.equal(1);
+    expect(MyMockContract.smocked.myFunction.calls[0]).to.deep.equal(['Something', 123]);
+
+.. container:: code-explanation
+
+  After:
+
+  .. code-block:: typescript
+
+    expect(myFakeContract.myFunction).to.be.calledOnceWith('Something', 123);
```

Comment:
Does this support deep assertions (if I return an array for instance)?

Reply:
Yes it does, it uses lodash `isEqualWith` with some customizers for bignumbers and structs